### PR TITLE
go/store/nbs: Fix table_index for table files with so many chunks that certain index slice operations overflow a uint32.

### PR DESCRIPTION
**Pull request by reltuk**

### CODE REVIEW

File: go/store/nbs/table_index_test.go [hunk 1]
```diff
@@ -56,6 +56,62 @@ func TestParseTableIndex(t *testing.T) {
...
+	}
+
+	// Write Lengths
+	for i := uint32(0); i < numChunks; i++ {
+		binary.BigEndian.PutUint32(idxBuf[off:], 4*1024)
+		off += lengthSize
+	}
+
+	// Write Suffixes
+	for i := uint32(0); i < numChunks; i++ {
```
Comment (macneale4) [inline go/store/nbs/table_index_test.go:87]: The three 320M iteration loops raises my eyebrows. How long does this take?

**Review by macneale4 — APPROVED** [review]
Behavior change makes sense. Just curious how long the test run is.

**Comment by coffeegoddd** [comment]
@reltuk __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>a876adb</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>a876adb</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

File: go/store/nbs/table_index_test.go (hunk 1 above)
Comment (nicktobey) [reply go/store/nbs/table_index_test.go:87]: And how much memory does this use? Can this be precomputed and checked in?

**Comment by coffeegoddd** [comment]
@coffeegoddd __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>47e0ce7</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>47e0ce7</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

File: go/store/nbs/table_index_test.go (hunk 1 above)
Comment (reltuk) [reply go/store/nbs/table_index_test.go:91]: It takes a few seconds to run. I'm not sure how much memory it uses. Somewhere between 4GB and 8GB.

**Comment by coffeegoddd** [comment]
@reltuk __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>d86a61c</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>d86a61c</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@coffeegoddd __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>76cd8a0</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>76cd8a0</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Pull request merged**